x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -162,21 +170,40 @@ protected Object unwrapFieldsMultiValue(Object values) {
                 if (isPrimitive(list) == false) {
                     if (list.size() == 1 || arrayLeniency) {
                         return unwrapFieldsMultiValue(list.get(0));
+                    } else if (allowMultiValue) {
```

Comment:
I'm wondering why you need a new `allowMultiValue` if the infrastructure already has `arrayLeniency`. Can't the latter be used instead? I feel like both are doing the same thing: allowing multi-value fields at extraction time. `arrayLeniency` is SQL specific and it maps to a REST request parameter, but can't we set `arrayLeniency` from EQL as well? When creating a new FieldHitExtractor is it possible to set this parameter to `true` and use the already existent code? (thinking out loud here, I cannot tell just by reading the code if it would work).

Reply:
Unfortunately the behavior of `arrayLeniency` is slightly different from what we need here:

```
[a, b] with no leniency =      error
[a, b] with arrayLeniency =    a
[a, b] with allowMultivalues = [a, b]
```

I could still use a boolean to express this three-way logic (true, false, null), but it would make the code much less readable.
Probably the best solution is to define an enum that better explains the intention, eg. 

```
AbstractFieldHitExtractor.MultivalueSupport.NOTE/LENIENT/FULL
```

Reply:
+1 for the enum